source/documentation/how-to/hosted-engine.html.md

```diff
@@ -346,3 +346,81 @@ If your hosted engine install fails, you have to manually clean up before you ca
        echo "! error removing $d"
        exit 1
     done
+
+### Purging Ovirt and supporting packages to allow Clean Reinstall
+The following is an example ansible script snippet that should clean the system so that no cruft remains to interfere wiht a new install.
+
+<!-- -->
+
+    - name: Clean Old Install
+      #This attempts to remove all old cruft from previous install attempts
+      #The reason we include the ovirt packages is so that they can be reinstall
+      #At potentially newer versions along with dependency packages
+      block:
+        - name: Detect existing cleanup script
+          shell: which ovirt-hosted-engine-cleanup | cat
+          register: ohes_cleanup
+        - name: Debug ohes_cleanup.stdout
+          debug:
+            var: ohes_cleanup.stdout
+        - name: Run Ovirt's Hosted Engine Cleanup Script
+          shell: ovirt-hosted-engine-cleanup -q
+          when: ohes_cleanup.stdout != ""
+        - name: Clean old packages
+          package:
+            name: "{{item}}"
+            state: absent
+          with_items:
+            - "*vdsm*"
+            - "*ovirt*"
+            - "*libvirt*"
+            - "*cockpit*"
+        - name: Remove old configs etc
+          shell: "rm -rf /etc/{{item}}"
+          args:
+            warn: False
+          with_items:
+            - "/etc/*ovirt*"
+            - "/etc/*vdsm*"
+            - "/etc/libvirt/qemu/HostedEngine*"
+            - "/etc/*libvirt*"
+            - "/etc/guacamole"
+            - "/etc/pki/vdsm"
+            - "/etc/pki/libvirt"
+            - "/etc/pki/CA"
+            - "/etc/pki/keystore"
+            - "/etc/ovirt-hosted-engine"
+            - "/var/lib/libvirt/"
+            - "/var/lib/vdsm/"
+            - "/var/lib/ovirt-hosted-engine-*"
+            - "/var/log/ovirt-hosted-engine-setup/"
+            - "/var/cache/libvirt/"
+            - "/etc/libvirt/nwfilter/vdsm-no-mac-spoofing.xml"
+            - "/var/cache/tomcat/temp/*"
+            - "/var/cache/tomcat/work/Catalina/localhost/*"
```

Comment:
Why tomcat?

Reply:
Sorry about that...I thought I cleaned out all the cruft from my specific special software
I must have missed tomcat.

I'll edit one more time.